networkx/utils/backends.py

```diff
@@ -315,333 +363,295 @@ def wrapper(*args, **kwds):
                 ]
             else:
                 graph = bound.arguments[gname]
                 if graph is None:
-                    if gname in optional_graphs:
+                    if gname in self.optional_graphs:
                         continue
                     raise TypeError(
-                        f"Missing required graph argument `{gname}` in {name} function"
+                        f"Missing required graph argument `{gname}` in {self.name} function"
                     )
-                if isinstance(_preserve_edge_attrs, dict):
+                if isinstance(preserve_edge_attrs, dict):
                     preserve_edges = False
-                    edges = _preserve_edge_attrs.get(gname, _edge_attrs)
+                    edges = preserve_edge_attrs.get(gname, edge_attrs)
                 else:
-                    preserve_edges = _preserve_edge_attrs
-                    edges = _edge_attrs
-                if isinstance(_preserve_node_attrs, dict):
+                    preserve_edges = preserve_edge_attrs
+                    edges = edge_attrs
+                if isinstance(preserve_node_attrs, dict):
                     preserve_nodes = False
-                    nodes = _preserve_node_attrs.get(gname, _node_attrs)
+                    nodes = preserve_node_attrs.get(gname, node_attrs)
                 else:
-                    preserve_nodes = _preserve_node_attrs
-                    nodes = _node_attrs
-                if isinstance(_preserve_graph_attrs, set):
-                    preserve_graph = gname in _preserve_graph_attrs
+                    preserve_nodes = preserve_node_attrs
+                    nodes = node_attrs
+                if isinstance(preserve_graph_attrs, set):
+                    preserve_graph = gname in preserve_graph_attrs
                 else:
-                    preserve_graph = _preserve_graph_attrs
+                    preserve_graph = preserve_graph_attrs
                 bound.arguments[gname] = backend.convert_from_nx(
                     graph,
                     edge_attrs=edges,
                     node_attrs=nodes,
                     preserve_edge_attrs=preserve_edges,
                     preserve_node_attrs=preserve_nodes,
                     preserve_graph_attrs=preserve_graph,
-                    name=name,
+                    name=self.name,
                     graph_name=gname,
+                    is_testing=is_testing,
                 )
 
-        result = getattr(backend, name).__call__(*bound.args, **bound.kwargs)
+        try:
+            result = getattr(backend, self.name)(*bound.args, **bound.kwargs)
+        except (NotImplementedError, NetworkXNotImplemented) as exc:
+            if fallback_to_nx:
+                return self.orig_func(*args, **kwargs)
+            if is_testing:
+                import pytest
+
+                pytest.xfail(
+                    exc.args[0] if exc.args else f"{name} raised {type(exc).__name__}"
```

Comment:
```suggestion
                    exc.args[0] if exc.args else f"{self.name} raised {type(exc).__name__}"
```

Reply:
Thanks. Included in latest commit. Black would rewrite your suggestion.